include/ck/tensor_operation/gpu/device/device_layernorm.hpp

```diff
@@ -0,0 +1,286 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <iostream>
+#include <sstream>
+
+#include "ck/utility/reduction_operator.hpp"
+#include "ck/tensor_operation/gpu/device/device_base.hpp"
+#include "ck/tensor_operation/gpu/device/device_reduce.hpp"
+#include "ck/tensor_operation/gpu/device/device_reduce_multiblock.hpp"
+#include "ck/tensor_operation/gpu/device/device_reduce_common.hpp"
+#include "ck/tensor_operation/gpu/grid/gridwise_layernorm.hpp"
+#include "ck/tensor_operation/gpu/grid/gridwise_set_buffer_value.hpp"
+#include "ck/device_utility/device_prop.hpp"
+#include "ck/device_utility/kernel_launch.hpp"
+
+namespace ck {
+namespace tensor_operation {
+namespace device {
+
+template <typename XDataType,
```

Comment:
We could use some consistent naming—we are complicating things by having X/Y data type here and In/Out vector size there

Reply:
eb6405ee5db9f538421a88adbcab26a277243c7e

Reply:
I am fine with the X/Y naming. Are we trending toward replacing In/Out terminology with X/Y for all other files?

cc @asroy

Reply:
I'm also fine with X/Y naming, as long as a clear math formula is given in the comment

Reply:
@rosenrodt  In case of layernorm, I don't use in because not only x is input, gamma and beta is also inputs